src/main/java/com/webperside/courseerpbackend/filters/AuthorizationFilter.java

```diff
@@ -26,27 +32,37 @@ public class AuthorizationFilter extends OncePerRequestFilter {
 
     private final AccessTokenManager accessTokenManager;
     private final AuthBusinessService authBusinessService;
+    private final RequestDataStorage requestDataStorage;
+    private final UserService userService;
 
     @Override
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
             throws ServletException, IOException {
 
         String token = request.getHeader(HttpHeaders.AUTHORIZATION);
 
+        String email = accessTokenManager.getEmail(
+                token.substring(7)
+        );
+
         try {
             if (Objects.nonNull(token) && token.startsWith(PREFIX)) {
+
                 authBusinessService.setAuthentication(
-                        accessTokenManager.getEmail(
-                                token.substring(7)
-                        )
+                        email
                 );
+
+                requestDataStorage.setUserId(userService.getByEmail(email).getId());
```

Comment:
setAuthentication method already returns UserDetails. so you can use it for getting user id

Reply:
I saw your comment here. 
as you know we have a class LoggedInUserDetails on the project (I can have a typo in the class name). So it includes id field as far as I know.
please try to use it

Reply:
@hamidsultanzadeh LoggedInUserDetails also doesn't have id field also

Reply:
then add it 😄